spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```diff
@@ -523,7 +518,13 @@ object DistriOptimizer {
     optimMethod: OptimMethod[T]
     )(implicit ev: TensorNumeric[T]) = {
     val sc = dataset.originRDD().sparkContext
-    val broadcast = sc.broadcast((model, criterion, state, validationMethods, optimMethod))
+    val broadcast = sc.broadcast((criterion, state, validationMethods, optimMethod))
+    // As cloneModel is using Serialization to implement deep copy, and will throw OOMError
+    // when model's size is bigger than SerializationUtils' buffer size. So we can use
+    // ModelBroadcast to clone model here.
+    // Notes: All models returned by modelBroadcast.value() share the same weight&bias, while
+    // gradWeight&gradBias is unshared.
+    val modelBroadcast = ModelBroadcast[T](false).broadcast(sc, model)
```

Comment:
Will it be OOM when broadcast weightAndBias?

Reply:
I think it's a bug of cloneModule. We should fix it or at least add a warning in the document of the method